srml/grandpa/src/lib.rs

```diff
@@ -376,3 +378,53 @@ impl<T: Trait> finality_tracker::OnFinalizationStalled<T::BlockNumber> for Modul
 		<Stalled<T>>::put((further_wait, median));
 	}
 }
+
+// TODO [slashing]: Integrate this.
+/// A grandpa equivocation offence report.
+#[allow(dead_code)]
+struct GrandpaEquivocationOffence<FullIdentification> {
+	/// A round in which the incident happened.
+	round: u64,
+	/// The session index that starts an era in which the incident happened.
+	current_era_start_session_index: u32, // TODO [slashing]: Should be a SessionIndex.
```

Comment:
it doesn't need to indicate the start of an era exactly; just the session where the offence occurred.

Reply:
Yeah, we've figured out that historical module will actually store the trie root for every session, so that should be renamed. CC @pepyakin

Reply:
Fixed!